tests/data_context/test_data_context_datasources.py

```diff
@@ -140,51 +276,109 @@ def test_delete_datasource_updates_cache(
     ) as mock_delete:
         context.delete_datasource(name)
 
-    assert not mock_delete.called
+    mock_delete.assert_called_once()
     assert name not in context.datasources
 
 
 @pytest.mark.unit
-def test_get_datasource_retrieves_from_cache(
+def test_BaseDataContext_add_datasource_updates_cache(
+    in_memory_runtime_context: BaseDataContext,
+) -> None:
+    """
+    What does this test and why?
+
+    For persistence-disabled contexts, we should only update the cache upon adding a
+    datasource.
+    """
+    context = in_memory_runtime_context
+
+    name = "my_new_datasource"
+
+    assert name not in context.datasources
+
+    context.add_datasource(
+        name,
+        class_name="PandasDatasource",
+        batch_kwargs_generators={
+            "subdir_reader": {
+                "class_name": "SubdirReaderBatchKwargsGenerator",
+            }
+        },
+    )
+
+    assert name in context.datasources
+
+
+@pytest.mark.unit
+def test_BaseDataContext_update_datasource_updates_existing_value_in_cache(
     in_memory_runtime_context: BaseDataContext,
 ) -> None:
+    """
+    What does this test and why?
+
+    For persistence-disabled contexts, we should only update the cache upon
+    updating an existing datasource.
+    """
     context = in_memory_runtime_context
 
     name = context.list_datasources()[0]["name"]
+    boto3_options = {"my": "option"}
+    datasource = PandasDatasource(name=name, boto3_options=boto3_options)
 
-    # If the value is in the cache, no store methods should be invoked
-    with mock.patch(
-        "great_expectations.data_context.store.DatasourceStore.get"
-    ) as mock_get:
-        context.get_datasource(name)
+    assert name in context.datasources
 
-    assert not mock_get.called
+    # Ensure that our cache value is updated to reflect changes
+    context.update_datasource(datasource)
+
+    assert name in context.datasources
+
+    retrieved_datasource = context.get_datasource(datasource_name=name)
+    assert retrieved_datasource.boto3_options == boto3_options
 
 
 @pytest.mark.unit
-def test_get_datasource_cache_miss(
+def test_BaseDataContext_update_datasource_creates_new_value_in_cache(
     in_memory_runtime_context: BaseDataContext,
 ) -> None:
+    """
+    What does this test and why?
+
+    For persistence-disabled contexts, we should only update the cache upon
+    using update to create a new datasource.
+    """
     context = in_memory_runtime_context
 
-    name = "my_fake_datasource_name"
+    name = "my_new_datasource"
+    boto3_options = {"my": "option"}
```

Comment:
Nit but you can skip the boto3_options in this test

Reply:
fixed!